pubsub/api/test/ConsoleOutputReader.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using System.IO;
+
+class ConsoleOutputReader : IDisposable
+{
+  TextWriter _STDOUT;
+  StringWriter _output = new StringWriter();

```

Comment:
As a developer, I would expect to see `System.Console.WriteLine()` used in code samples/snippets.

If the sample function accepts a StringWriter to write to, that:
 1. Violates the principle of least surprise
 2. Adds additional complexity to the sample

Existing Java samples use `System.out.println` and Python samples use `print` which are intuitive to developers familiar with those languages.